src/main/java/frc/robot/subsystems/swerve/swervestatehelpers/AimAssistMath.java

```diff
@@ -67,4 +67,17 @@ public static double applyMagnitudeCompensation(Rotation2d velocityPerSecond, do
 			/ (magnitude + SwerveConstants.AIM_ASSIST_MAGNITUDE_FACTOR);
 	}
 
+	public static Translation2d getClosestCutPoint(Translation2d robot, Translation2d speaker, double ShootingZoneRadius){
+		double m = robot.getY() / robot.getX();
+		double cutPointX = Math.sqrt(
+				Math.pow(ShootingZoneRadius,2)
+				/ (Math.pow(m,2) + 1)
+		);
+		if (robot.getX() < 0) {
+			cutPointX *= -1;
+		}
```

Comment:
will not work (imagine if the speaker of middle of field)

Reply:
this calculation is normalizing it to thinking the speaker is it 0,0 and then just adding the center x,y. why wont it work?

Reply:
why <0

Reply:
you act like the robot always have a bigger x than the target

Reply:
The result x of the calculation will always be positive, so you check if the origin x was negative because if it was u want to keep it that way